ogre/src/OgreLidarVisual.cc

```diff
@@ -275,41 +342,105 @@ void OgreLidarVisual::Update()
       ignition::math::Vector3d noHitPt =
                   (axis * noHitRange) + this->offset.Pos();
 
-      // Draw the lines and strips that represent each simulated ray
-      if (i >= this->dataPtr->rayLines[j]->PointCount()/2)
-      {
-        this->dataPtr->rayLines[j]->AddPoint(startPt);
-        this->dataPtr->rayLines[j]->AddPoint(inf ? noHitPt : pt);
-
-        this->dataPtr->rayStrips[j]->AddPoint(startPt);
-        this->dataPtr->rayStrips[j]->AddPoint(inf ? startPt : pt);
+      // Update the lines and strips that represent each simulated ray.
 
-        this->dataPtr->noHitRayStrips[j]->AddPoint(startPt);
-        this->dataPtr->noHitRayStrips[j]->AddPoint(inf ? noHitPt : pt);
-      }
-      else
+      // For RAY_LINES Lidar Visual to be displayed
+      if (this->dataPtr->lidarVisType ==
+                LidarVisualType::LVT_RAY_LINES)
       {
-        this->dataPtr->rayLines[j]->SetPoint(i*2, startPt);
-        this->dataPtr->rayLines[j]->SetPoint(i*2+1, inf ? noHitPt : pt);
+        if (i >= this->dataPtr->rayLines[j]->PointCount()/2)
+        {
+          if (this->displayNonHitting || !inf)
+          {
+            this->dataPtr->rayLines[j]->AddPoint(startPt);
+            this->dataPtr->rayLines[j]->AddPoint(inf ? noHitPt : pt);
+          }
+        }
+        else
+        {
+          if (this->displayNonHitting || !inf)
+          {
+            this->dataPtr->rayLines[j]->SetPoint(i*2, startPt);
+            this->dataPtr->rayLines[j]->SetPoint(i*2+1, inf ? noHitPt : pt);
+          }
+        }
+      }
 
-        this->dataPtr->rayStrips[j]->SetPoint(i*2, startPt);
-        this->dataPtr->rayStrips[j]->SetPoint(i*2+1, inf ? startPt : pt);
+      // For TRIANGLE_STRIPS Lidar Visual to be displayed
+      else if (this->dataPtr->lidarVisType ==
+                LidarVisualType::LVT_TRIANGLE_STRIPS)
```

Comment:
The Update function is getting long with a bit of duplicate code. Can the logic for triangle strips be merged with lines since the strips visualization method also includes drawing lines? 

Same for the code for setting materials above. We could refactor that out to a different function if that's possible.

Reply:
Can you please check the updated code. I have somewhat merged it together.

Reply:
yep that looks fine